app/index.ts

```diff
@@ -31,57 +34,94 @@ if (!existsSync('.env')) {
 // Setup command line options
 program
     .name('saiki')
-    .description('AI-powered CLI for interacting with MCP servers')
+    .description('AI-powered CLI and WebUI for interacting with MCP servers')
     .option('-c, --config-file <path>', 'Path to config file', 'configuration/saiki.yml')
     .option('-s, --strict', 'Require all server connections to succeed')
     .option('--no-verbose', 'Disable verbose output')
-    .version('0.1.0');
+    .option('--mode <mode>', 'Run mode: cli, web, or both', 'cli')
+    .option('--web-port <port>', 'Port for WebUI', '3000')
+    .version('0.2.0');
 
 program.parse();
 
 // Get options
 const options = program.opts();
 const configFile = options.configFile;
 const connectionMode = options.strict ? 'strict' : ('lenient' as 'strict' | 'lenient');
+const runMode = options.mode.toLowerCase();
+const webPort = parseInt(options.webPort, 10);
+
+// Validate run mode
+if (!['cli', 'web', 'both'].includes(runMode)) {
+    logger.error(`Invalid mode: ${runMode}. Must be 'cli', 'web', or 'both'.`);
+    process.exit(1);
+}
+
+// Validate web port
+if (isNaN(webPort) || webPort <= 0 || webPort > 65535) {
+    logger.error(`Invalid web port: ${options.webPort}. Must be a number between 1 and 65535.`);
+    process.exit(1);
+}
 
 // Platform-independent path handling
 const normalizedConfigPath = path.normalize(configFile);
 
 logger.info(
-    `Starting AI-powered MCP client with config file: ${normalizedConfigPath}`,
+    `Starting Saiki Agent with config: ${normalizedConfigPath}`,
     null,
     'blue'
 );
 
-// Display examples
-logger.info('');
-logger.info(
-    'This client uses LLM models to interpret your commands and call appropriate MCP tools.',
-    null,
-    'cyanBright'
-);
-logger.info('You can interact with tools using natural language.');
-logger.info('');
-logger.info('Examples:', null, 'yellow');
-logger.info('- "List all files in the current directory"');
-logger.info('- "Show system information"');
-logger.info('- "Create a new file called test.txt with \'Hello World\' as content"');
-logger.info('- "Run a simple python script that prints numbers from 1 to 10"');
-logger.info('');
-
-// Start the AI client directly
-async function startAiClient() {
+// Conditionally display CLI examples
+if (runMode === 'cli' || runMode === 'both') {
```

Comment:
we don't need this condition, we can display it anyway. it won't show up in web ui either way